Updates to our releasing docs (#3121)
alphagov/govuk-frontend: docs/releasing/before-publishing-a-release.md

# Tasks before publishing a new release of GOV.UK Frontend

When preparing to publish a release of GOV.UK Frontend we need to ensure we are following proper procedure to effectively track our work and are prepared to publish our new release.

## Kick off the release

> Both squads should coordinate when either or both wants to publish a new release. The first step of this should be to select who will lead on the release, typically a developer.

Reviewer: ```suggestion
Both squads should coordinate when either wants to publish a new release. Choose a team member to lead on the release, typically a developer.
```

  - bump the version of `govuk-frontend`
  - update the "Updating with npm" example for `package.json` with the current number
  - merge any other documentation PRs specific to the release
- post the comms and do tidy-up tasks (example issue: [#2510](https://github.com/alphagov/govuk-frontend/issues/2510))

Once the developers have created these issues, the person leading the release should add them to an epic (example issue: [#2511](https://github.com/alphagov/govuk-frontend/issues/2511)). All these issues should be added to the [Design System sprint board](https://github.com/orgs/alphagov/projects/4).

## Draft comms and release notes for the community

A content designer and/or tech writer should do the following:
- write announcements for slack posts, email and to go on the design system website after we release GOV.UK Frontend (for example, [draft comms for the cookie banner component](https://docs.google.com/document/d/1jVyMB7i94NOeflWaf3kE4Q4APMXGfluK3rOh74IHO08/edit))
- check who the release’s contributors are and if we have consent to include their name

> A technical writer should finalise draft of release notes and get 2i on them. If the technical writer is unavailable, ask for help in the [gds-technical-writing Slack channel](https://gds.slack.com/archives/CAD0R2NQG) or confer with a content designer.

Reviewer: Be helpful to add enough time for tech writer 2i to take place.

```suggestion
A technical writer should finalise draft of release notes. Release notes will require a 2i review by another technical writer, make sure the technical writer has time to coordinate this.

If the technical writer is unavailable, ask for help in the [gds-technical-writing Slack channel](https://gds.slack.com/archives/CAD0R2NQG) or confer with a content designer.
```

> The team should also update relevant [issue discussions](https://github.com/orgs/alphagov/projects/43/views/1) with rationale for any decisions we've made.

Reviewer: ```suggestion
The team should also post an message on any relevant [issue discussions](https://github.com/orgs/alphagov/projects/43/views/1) with rationale for any decisions we've made.
```
